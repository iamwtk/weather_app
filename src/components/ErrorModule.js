import React from 'react'



import style from './errorModule.scss'

const ErrorModule = (props) => (
  <div className={style.error_module}>
  <p>Sorry, there was an error!</p>
  {props.errors.map((err, i) => <span key={i}>{err.message}</span>)}
  </div>
);

export default ErrorModule;
