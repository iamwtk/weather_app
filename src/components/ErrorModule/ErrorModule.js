import React, { PropTypes } from 'react'

import style from './errorModule.scss'

/**
 * <ErrorModule />
 * displays errors from array passed as props
 * @param {array} errors array of errors occured while fetching
 */
const ErrorModule = ({errors}) => (
  <div className={style.error_module}>
    <p>Sorry, there was an error!</p>
    {errors.map((err, i) => <span key={i}>{err.message}</span>)}
  </div>
)
export default ErrorModule
