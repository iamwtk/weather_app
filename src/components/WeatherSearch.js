import React from 'react'
import PlacesAutocomplete from 'react-places-autocomplete'
import { Link } from 'react-router-dom';

import ErrorModule from './ErrorModule';

import SearchIcon from '-!svg-react-loader?name=MyIcon!../icons/searchIcon.svg'

import style from './weatherSearch.scss';

const WeatherSearch = (props) => (
  <div className={style.search_container}>
    <div id='search_bar' className={style.search_bar}>
      <form onSubmit={props.handle} className={style.form}>
        <PlacesAutocomplete
        inputProps={props.inputProps}
        options={props.options}/>
        <button type="submit" className={style.search_button}><SearchIcon /></button>
      </form>
      { props.errors.length ? <ErrorModule errors={props.errors} /> : null}
    </div>
  </div>
)

export default WeatherSearch
