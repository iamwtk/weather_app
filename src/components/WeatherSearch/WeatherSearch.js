import React from 'react'
import PlacesAutocomplete from 'react-places-autocomplete'
import { Link } from 'react-router-dom'

import ErrorModule from '../ErrorModule/ErrorModule'

import SearchIcon from '-!svg-react-loader?name=MyIcon!../../icons/searchIcon.svg'

import style from './weatherSearch.scss'


/**
 * WeatherSearch Component - shows search bar form
 * @param {Object} props passed by parent component
 */
const WeatherSearch = (props) => {
  //Destructuring of props
  const { handle, inputProps, options, errors } = props
  return (
    <div className={style.search_container}>
      <div id='search_bar' className={style.search_bar}>
        <form onSubmit={handle} className={style.form}>
          <PlacesAutocomplete
          inputProps={inputProps}
          options={options}/>
          <button type="submit" className={style.search_button}><SearchIcon /></button>
        </form>
        { errors.length ? <ErrorModule errors={errors} /> : null}
      </div>
    </div>
  )
}
export default WeatherSearch
