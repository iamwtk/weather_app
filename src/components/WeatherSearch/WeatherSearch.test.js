import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';

import PlacesAutocomplete from 'react-places-autocomplete'
import WeatherSearch from './WeatherSearch'
import ErrorModule from '../ErrorModule/ErrorModule'

import style from './weatherSearch.scss'

describe('<WeatherSearch/>', () => {
  let component
  let errors = []
  const options = {}
  const handle = jest.fn()
  describe('Form', () => {
    it('Should call handle() on form submit', () => {
      component = shallow(<WeatherSearch handle={handle} errors={errors}/>)
      component.find('form').simulate('submit')
      expect(handle).toBeCalled()
    })
  })
  describe('Errors', () => {
    describe('No errors', () => {
      it('If no errors component <ErrorModule/> should not render', () => {
          expect(component.contains(<ErrorModule errors={errors}/>)).toBe(false)
      })
      it('should match the snapshot', () => {
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
      })
    })
    describe('Has errors', () => {
      it('If errors component should render <ErrorModule/> with errors as props', () => {
        errors = [{message: 'error 1'}]
        component = shallowu(<WeatherSearch handle={handle} errors={errors}/>)
        expect(component.contains(<ErrorModule errors={errors}/>)).toBe(true)
      })
      it('should match the snapshot', () => {
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
      })
    })
  })

});
