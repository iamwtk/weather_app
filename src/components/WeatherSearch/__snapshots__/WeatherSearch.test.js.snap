// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherSearch/> Errors Has errors should match the snapshot 1`] = `
<div
  className="search_container"
>
  <div
    className="search_bar"
    id="search_bar"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <PlacesAutocomplete
        autocompleteItem={[Function]}
        classNames={Object {}}
        clearItemsOnError={false}
        debounce={200}
        googleLogo={true}
        googleLogoType="default"
        highlightFirstSuggestion={false}
        onError={[Function]}
        options={Object {}}
        styles={Object {}}
      />
      <button
        className="search_button"
        type="submit"
      />
    </form>
  </div>
</div>
`;

exports[`<WeatherSearch/> Errors No errors should match the snapshot 1`] = `
<div
  className="search_container"
>
  <div
    className="search_bar"
    id="search_bar"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <PlacesAutocomplete
        autocompleteItem={[Function]}
        classNames={Object {}}
        clearItemsOnError={false}
        debounce={200}
        googleLogo={true}
        googleLogoType="default"
        highlightFirstSuggestion={false}
        onError={[Function]}
        options={Object {}}
        styles={Object {}}
      />
      <button
        className="search_button"
        type="submit"
      />
    </form>
  </div>
</div>
`;
